wallet/db_sqlite3.c

```diff
@@ -146,6 +146,12 @@ static bool db_sqlite3_setup(struct db *db)
 	}
 	wrapper->conn = sql;
 
+	err = sqlite3_extended_result_codes(wrapper->conn, 1);
```

Comment:
Seems to me like this code is unreachable iff `sqlite3_extended_result_codes(...) != SQLITE_OK => sqlite3_open_v2(...)` which seems to be logically the case, since otherwise `sqlite3_open_v2` would not fail if the extended error returns an error (which would mean `sqlite3_open_v2` is allowed to fail silently...)

So might be better to pull the extended error up above line 141.

Reply:
My bad, seems I misunderstood the semantics of the call (I was thinking it'd return more details on a current error, not enable future error to be more verbose).